histomicstk/features/compute_global_graph_features.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import division
+
+from collections import namedtuple
+from itertools import combinations
+
+import numpy as np
+from numpy import linalg
+from scipy.spatial import cKDTree as KDTree, Voronoi
+from scipy import sparse
+from scipy.sparse.csgraph import minimum_spanning_tree
+
+PopStats = namedtuple('PopStats', ['mean', 'stddev', 'minmaxr', 'disorder'])
+PolyProps = namedtuple('PolyProps', ['area', 'peri', 'max_dist'])
+TriProps = namedtuple('TriProps', ['sides', 'area'])
+DensityProps = namedtuple('DensityProps', ['neighbors_in_distance',
+                                           'distance_for_neighbors'])
+Props = namedtuple('Props', ['voronoi', 'delaunay', 'mst_branches', 'density'])
+
+
+def compute_global_graph_features(centroids, scale=10., neighbor_counts=(3, 5, 7)):
+    """Compute global (i.e., not per-pixel) graph-based features of the
+    nuclei with the given centroids.
+
+    Parameters
+    ----------
+    centroids : array_like
+        Nx2 numpy array of nuclear centroids
+    scale : float
+        Multiplier for distances.  Used for determining the radii to
+        count neighbors in.
+    neighbor_counts : sequence
+        Sequence of numbers of neighbors, each of which is used to
+        compute statistics relating to the distance required to reach
+        that many neighbors.
+
+    Returns
+    -------
+    props : collections.namedtuple
+        Nested namedtuples with the structure:
+
+        - .voronoi: Voronoi diagram features
+
+          - .area: Polygon area features
+          - .peri: Polygon perimeter features
+          - .max_dist: Maximum distance in polygon features
+
+        - .delaunay: Delaunay triangulation features
+
+          - .sides: Triangle side length features
+          - .area: Triangle area features
+
+        - .mst_branches: Minimum spanning tree branch features
+        - .density: Density features
+
+          - .neighbors_in_distance
+
+            - [radius]: Neighbor count within given radius features
+
+          - .distance_for_neighbors
+
+            - [count]: Minimum distance to enclose count neighbors features
+
+        Each leaf node is itself a namedtuple with fields 'mean',
+        'stddev', 'minmaxr', and 'disorder'.  'minmaxr' is the
+        minimum-to-maximum ratio, and disorder is stddev / (mean +
+        stddev).
```

Comment:
Return a pandas dataframe instead of a nest named tuple similar to other feature computation functions. 

Ultimately we will concat different types of features which will be easy if all of them return in the same format. Otherwise, client code of this function will have to write code to convert this one to a pandas dataframe